bakery/service_test.go

```diff
@@ -131,10 +143,93 @@ func (s *ServiceSuite) TestMacaroonPaperFig6FailsWithBindingOnTamperedSignature(
 	}
 
 	// client makes request to ts.
-	err = ts.Check(d, strCompFirstPartyChecker(""))
+	err = ts.Check(d, strcmpChecker(""))
 	c.Assert(err, gc.ErrorMatches, "verification failed: signature mismatch after caveat verification")
 }
 
+func (s *ServiceSuite) TestNeedDeclared(c *gc.C) {
```

Comment:
The nested third-party caveat need-declared thing is an interesting possibility, but one that can be added later, I think. I've written another test to check with two third party discharges. I think the last example is sufficiently tested by the existing lower level checks in checkers.

With nested need-declared, the inner one won't be recognised - it's recognised directly by the bakery third party checker and nowhere else.